src/Executor/ReferenceExecutor.php

```diff
@@ -601,6 +601,10 @@ protected function resolveField(ObjectType $parentType, $rootValue, \ArrayObject
             return static::$UNDEFINED;
         }
 
+        if (! $fieldDef->isVisible($contextValue)) {
+            return static::$UNDEFINED;
```

Comment:
So for clients, the field would simply not show up in the results? I am pretty sure this behaviour goes against the GraphQL specification. If anything, I would expect a validation error.

Reply:
I wasn't really sure how to proceed here since the spec is not clear on this case. If we prefer validation instead, where would be the best place to add a new test case?

Reply:
Probably next to the test cases that ensure that an error is thrown when a client requests entirely unknown fields. I would expect hidden fields to behave exactly the same as a field that is defined at all, at least from a client perspective.